src/interfaces.ts

```diff
@@ -13,3 +15,10 @@ export interface IBLEDevice extends EventEmitter {
     readFromCharacteristic: (uuid: string, callback: (err: string | null, data: Buffer | null) => void) => void;
     writeToCharacteristic: (uuid: string, data: Buffer, callback?: () => void) => void;
 }
+
+export interface IDeviceInterface extends EventEmitter {
+    type: Consts.HubType;
+    getPortNameForPortId: (portId: number) => string | undefined;
+    send: (message: Buffer, uuid: string, callback?: () => void) => void;
+    subscribe: (portId: number, deviceType: number, mode: number) => void;
+}
```

Comment:
Even if passed to devices, it seems to be a Hub interface.

Reply:
Yeah - I've had to do this for the moment as TypeScript doesn't like circular dependencies/definitions - Hub imports Device which imports Hub (even if for the type only). This was a quick fix - I plan to revisit this.

Reply:
My concern was more about the naming, I took it for a device definition at first.

Reply:
Makes sense - will change